meson.options

```diff
@@ -0,0 +1,30 @@
+option(
+  'tests',
+  type: 'feature',
+  value: 'auto',
+  description: 'Build tests',
+)
+option(
```

Comment:
Avoid splitting `tests_freestanding` from `tests`. Keep a single tests option and gate freestanding internally if needed.